src/storage/store/var_list_column.cpp

```diff
@@ -11,13 +11,37 @@ using namespace kuzu::transaction;
 namespace kuzu {
 namespace storage {
 
-offset_t ListOffsetInfoInStorage::getListOffset(uint64_t nodePos) const {
-    if (nodePos == 0) {
-        return prevNodeListOffset;
-    } else {
-        auto offsetVector = offsetVectors[(nodePos - 1) / DEFAULT_VECTOR_CAPACITY].get();
-        return offsetVector->getValue<offset_t>((nodePos - 1) % DEFAULT_VECTOR_CAPACITY);
+offset_t ListOffsetInfoInStorage::getListStartOffset(uint64_t nodePos) const {
+    if (numTotal == 0) {
+        return 0;
     }
+    return nodePos == numTotal ? getListEndOffset(nodePos - 1) :
+                                 getListEndOffset(nodePos) - getListLength(nodePos);
+}
+
+offset_t ListOffsetInfoInStorage::getListEndOffset(uint64_t nodePos) const {
+    KU_ASSERT(nodePos < offsetVectors.size() * common::DEFAULT_VECTOR_CAPACITY);
+    auto offsetVector = offsetVectors[(nodePos) / DEFAULT_VECTOR_CAPACITY].get();
+    return offsetVector->getValue<offset_t>((nodePos) % DEFAULT_VECTOR_CAPACITY);
+}
+
+uint64_t ListOffsetInfoInStorage::getListLength(uint64_t nodePos) const {
+    KU_ASSERT(nodePos < sizeVectors.size() * common::DEFAULT_VECTOR_CAPACITY);
+    auto sizeVector = sizeVectors[nodePos / common::DEFAULT_VECTOR_CAPACITY].get();
+    return sizeVector->getValue<uint64_t>(nodePos % common::DEFAULT_VECTOR_CAPACITY);
+}
+
+bool ListOffsetInfoInStorage::checkOffsetOder(uint64_t startPos, uint64_t endPos) const {
```

Comment:
Why not just check if end offsets are sorted ascending? This code should be much simplified if so.